Providers/nxOMSAutomationWorker/automationworker/3.x/worker/runtime.py

```diff
@@ -0,0 +1,212 @@
+                                                                      env=env,
+                                                                      stdout=subprocess.PIPE,
+                                                                      stderr=subprocess.PIPE)
+
+    def kill_runbook_subprocess(self):
+        """Attempts to kill the runbook subprocess.
+
+        This method will attempt to kill the runbook subprocess [max_attempt_count] and will return if successful.
+
+        Throws:
+            SandboxRuntimeException : If runbook subprocess is still alive after [max_attempt_count].
+        """
+        attempt_count = 0
+        max_attempt_count = 3
+        while attempt_count < max_attempt_count:
+            if self.runbook_subprocess is not None and self.runbook_subprocess.poll() is None:
+                os.kill(self.runbook_subprocess.pid, signal.SIGTERM)
+                runbook_proc_is_alive = self.is_process_alive(self.runbook_subprocess)
+                if runbook_proc_is_alive is False:
+                    return
+                attempt_count += 1
+                time.sleep(attempt_count)
+            else:
+                return
+        raise SandboxRuntimeException()
+
+    @staticmethod
+    def is_process_alive(process):
+        """Checks if the given process is still alive.
+
+        Returns:
+            boolean : True if the process [pid] is alive, False otherwise.
+        """
+        if process.poll() is None:
+            return True
+        else:
+            return False
+
+    def is_runtime_supported(self):
+        """Validates that the OS supports the language runtime by testing the executable file path.
+
+        Returns:
+            True    : If executable exist.
+            False   : Otherwise.
+        """
+        if find_executable(self.execution_alias) is None:
+            return False
+        else:
+            return True
+
+
+class PowerShellRuntime(Runtime):
+    """PowerShell runtime derived class."""
+
+    def __init__(self, job_data, runbook):
+        Runtime.__init__(self, job_data, runbook)
+
+        self.execution_alias = "pwsh"
+        if linuxutil.is_posix_host() is False:
+            self.execution_alias = "powershell"
+            
+        self.base_cmd = [self.execution_alias, "-File"]
+
+
+class Python2Runtime(Runtime):
+    """Python 2 runtime derived class."""
+
+    def __init__(self, job_data, runbook):
+        Runtime.__init__(self, job_data, runbook)
+        self.execution_alias = "python2"
+
+        if get_default_python_interpreter_major_version() == 2:
+            self.execution_alias = "python"
+
+        self.base_cmd = [self.execution_alias]
+
+
+class Python3Runtime(Runtime):
+    """Python 3 runtime derived class."""
+
+    def __init__(self, job_data, runbook):
+        Runtime.__init__(self, job_data, runbook)
+        self.execution_alias = "python3"
+
+        if get_default_python_interpreter_major_version() == 3:
```

Comment:
Redundant check, if  get_default_python_interpreter_major_version will always return 3 for this file.